README.md

```diff
@@ -1 +1,184 @@
+## Local development
+
+### Dependencies
+
+- [NodeJS](https://nodejs.org/en/download/)
+- [Yarn](https://classic.yarnpkg.com/en/docs/install/#debian-stable)
+- [Go](https://golang.org/doc/install)
+- [Hugo](https://gohugo.io/getting-started/installing/)
+
+If you're running the site for the first time, or if dependencies have changed, install dependencies with:
+
+`yarn install`
+
+### Pre-load Webpack assets
+
+Additionally, you may want to run:
+
+`npm run build:webpack`
+
+Hugo is so fast that the first time you run the site, the Hugo build will happen much more quickly than the webpack build so it's best to run the build once so that assets are ready for when you start the site.  If you don't see any styling the first time you visit the page, force refresh it.  If that doesn't work, try restarting the site.
+
+### External API's
+
+This site requires running instances of [`ocw-studio`](https://github.com/mitodl/ocw-studio) and [`open-discussions`](https://github.com/mitodl/open-discussions) for some functionality.  Search results are provided by `open-discusisons` and `ocw-studio` provides some content for the home page, such as newly added courses and news items.  If you need to work with this functionality you can either run a local instance of either of these projects, or alternatively point at the RC instances and use a browser extension to temporarily disable CORS.
+
+### CORS
+
+The search page at `/search` uses the `open-discussions` search API to source results.  Running this locally and populating it with results can be tedious, so it's often easier to just point your local website at an already running version of the search API.  In order for this to work properly, you will need to use a browser extension for disabling CORS, like Moesif Origin & CORS Changer:
+
+- [Chrome](https://chrome.google.com/webstore/detail/moesif-origin-cors-change/digfbfaphojjndkpccljibejjbppifbc?hl=en-US#:~:text=Moesif%20Origin%20%26%20CORS%20Changer&text=This%20plugin%20allows%20you%20to%20send%20cross%2Ddomain%20requests.&text=This%20plugin%20allows%20you%20to%20send%20cross%2Ddomain%20requests%20directly,Allow%2DOrigin%20set%20to%20*.)
+- [Firefox](https://addons.mozilla.org/en-US/firefox/addon/moesif-origin-cors-changer1/)
+### Environment variables
+
+For development you'll need to set a few environment variables. The
+`.env.example` file will get you started.  Here are the variables and what they do:
+
+| Variable | Example | Description |
+| --- | --- | --- |
+| `SEARCH_API_URL` | `http://localhost:8063/api/v0/search/` | A URL to an `open-discussions` search API to fetch results from|
+| `OCW_STUDIO_BASE_URL` | `http://localhost:8043/` | A URL of an instance of [`ocw-studio`](https://github.com/mitodl/ocw-studio) to fetch home page content from |
+| `EXTERNAL_SITE_PATH` | `~/Code/ocw-www/site/` | A path to a Hugo site with content (Pages, Notifications, Promos, Testimonials) to use for local development |
```

Comment:
If I'm building a single course, would I set `EXTERNAL_SITE_PATH` to the path of the course repo?

Reply:
Currently course sites can only be spun up using `ocw-to-hugo`, but I can add the ability to spin them up from `EXTERNAL_SITE_PATH` if `OCW_TO_HUGO_PATH` is not set.

Reply:
Actually, I just tested this and you can spin up a course site without `ocw-to-hugo`.  You do need to add a couple of files to your course content folder for it to work though:

`config/_default/config.toml`:

```
baseURL = "http://localhost:3000/"
languageCode = "en-us"
title = "MIT OpenCourseWare"

[module]
  [[module.imports]]
    path = "github.com/mitodl/ocw-hugo-theme/base-theme"
  [[module.imports]]
    path = "github.com/mitodl/ocw-hugo-theme/course"
  [[module.mounts]]
    source = "assets"
    target = "assets"
  [[module.mounts]]
    source = "data"
    target = "data"
  [[module.mounts]]
    source = "layouts"
    target = "layouts"
  [[module.mounts]]
    source = "static"
    target = "static"
```

`go.mod`:
```
module github.com/mitodl/1-00-introduction-to-computers-and-engineering-problem-solving-spring-2012

go 1.16

replace github.com/mitodl/ocw-hugo-theme/base-theme => /home/gumaerc/Code/ocw-hugo-theme/base-theme

replace github.com/mitodl/ocw-hugo-theme/course => /home/gumaerc/Code/ocw-hugo-theme/course
```

After placing these files in your course content folder and running `npm start` in the theme, a local dev site will spin up pointing at that course site content.  With the way our content currently comes through legacy OCW though, if you are developing changes to the theme it's much more simple to just have it use `ocw-to-hugo` to get the content.